packages/custom-storage/__tests__/SQLStorage.spec.ts

```diff
@@ -0,0 +1,179 @@
+
+describe('SQLStorage', () => {
+  let sqlStorage: SQLStorage;
+
+  beforeAll(() => {
+    const sqlClient = new SQL('user', 'm14p1k3y', 'https://{user}.carto.com/');
+    sqlStorage = new SQLStorage('mynamespace', sqlClient, 1, true);
+  });
+
+  it('should check or create tables for storage', async () => {
+    // We use only one mock because is not important for the rest of the queries
+    (global as any).fetch.mockResponse(
+      JSON.stringify({
+        rows: [
+          {rolename: 'cartodb_publicuser_a1b2c3d4f5'}
+        ],
+        time: 0.001,
+        fields: {
+          rolename: {type: 'name', pgtype: 'name'}
+        },
+        total_rows: 1
+      }),
+      {
+        headers: [
+          ['content-type', 'application/json'],
+          ['Retry-After', 0],
+          ['Carto-Rate-Limit-Remaining', 5]
+        ]
+      }
+    );
+    await sqlStorage.init();
+
+    // Check the creation and access grant queries of the 3 tables
+    (global as any).fetch.mock.calls.forEach((call: any, idx: number) => {
+      const params = (global as any).getURLParams(call[0]);
+      expect(decodeURI(params.q)).toBe(CHECK_CREATE_QUERIES[idx]);
+    });
+  });
+
+  it('should list visualizations', async () => {
+    const visObject: StoredVisualization = {
+      id: 'a1b2c3d4',
+      name: 'myVis',
+      description: 'This is a test vis',
+      thumbnail: '',
+      config: '{}',
+      lastModified: '2019-11-02T14:00:00Z',
+      isPrivate: false
+    };
+
+    (global as any).fetch.mockResponse(
+      JSON.stringify(
+        {
+          rows: [
+            {
+              id: 'a1b2c3d4',
+              name: 'myVis',
+              description: 'This is a test vis',
+              thumbnail: '',
+              config: {},
+              last_modified: '2019-11-02T14:00:00Z',
+              private: false
+            }
+          ],
+          time: 0.001,
+          fields: {
+            id: {type: 'string', pgtype: 'uuid'},
+            name: {type: 'string', pgtype: 'text'},
+            description: {type: 'string', pgtype: 'text'},
+            thumbnail: {type: 'string', pgtype: 'text'},
+            config: {type: 'string', pgtype: 'json'},
+            last_modified: {type: 'string', pgtype: 'date'},
+            private: {type: 'boolean', pgtype: 'bool'}
+          },
+          total_rows: 0
+        }
+      ),
+      {
+        headers: [
+          ['content-type', 'application/json'],
+          ['Retry-After', 0],
+          ['Carto-Rate-Limit-Remaining', 5]
+        ]
+      }
+    );
+
+    const storedVis = await sqlStorage.getVisualizations();
+
+    // Check list visualizations query
+    const lastCall = (global as any).fetch.mock.calls[(global as any).fetch.mock.calls.length - 1];
+    const params = (global as any).getURLParams(lastCall[0]);
+    expect(decodeURI(params.q)).toBe(LIST_VISUALIZATION_QUERY);
+
+    // Check response
+    expect(storedVis).toStrictEqual([visObject]);
```

Comment:
Here, in the second expect we are probably testing 'rowToVisualization' from utils, more than the SQLStorage. getVisualizations(). I think that the first check could probably be enough

Reply:
Yes, this tests could fit better for `rowToVisualization`; but I thought that it would be important to always check the returned value of a function to avoid issues related to changing its format. Anyway, I can move this test to an utils file if you prefer.